### ISSUE TITLE
macOS Silicon Native Build

### ISSUE THREAD

**Issue by aidan-gibson** [nmap/nmap#2942]
A native build of Zenmap for silicon macOS would be great thanks.

**Comment by fyodor** [comment]
Yes, a native Applie Silicon (ARM64) version of Nmap would be useful now and we will probably have to do it in the coming years anyway because Apple is threatening to start phasing out the Rosetta system that does the x64 compatibility translation.  So we have bumped up the priority of this.

The first step will be to decide whether to do just a clean ARM64 build or use the Universal Binary 2 system so we can still support the older Intel macs with the newer Nmap releases.  Otherwise the older macs will have to use an old Nmap version too.  But supporting both adds complexity and difficulty (plus bloats the size), so it will depend on what demand there really is for continued x64 support at that time.  It looks like Apple is planning to drop Intel support in their next major macOS release (macOS 27, due out this year).

The next step will be to set up our build systems to compile everything ARM64 (or Universal).  This will include Nmap, Ncat, and Nping themselves and dependencies, plus the python interpreter used for running Zenmap and its dependencies.

**Comment by LeMathochist** [comment]
No longer a threat. Every time I load anything that is translated through Rosetta, I get the warning. According to this [Apple Support ](https://support.apple.com/en-us/102527) page, MacOS 28 will be the end of Rosetta for anything but a handful of unmaintained games (of all things). So there's a little more time.
